packages/teleport/src/Player/Player.tsx

```diff
@@ -30,9 +30,21 @@ import { getUrlParameter } from 'teleport/services/history';
 export default function Player() {
   const { sid, clusterId } = useParams<UrlPlayerParams>();
   const recordingType = getUrlParameter('recordingType', useLocation().search);
+  const durationMs = parseFloat(
+    getUrlParameter('durationMs', useLocation().search)
+  );
+
   if (recordingType !== 'ssh' && recordingType !== 'desktop') {
     throw new Error(
-      `Invalid recording type: ${recordingType}, should be 'ssh' or 'desktop'`
+      `Invalid query parameter recordingType: ${recordingType}, should be 'ssh' or 'desktop'.`
+    );
+  }
+  if (isNaN(durationMs) || !Number.isInteger(durationMs)) {
+    throw new Error(
```

Comment:
Typically I prefer to have components gracefully fail and handle these errors internally instead of throwing. I see above that this component is already throwing so you might want to do this as a follow-up.

Reply:
Good call, I fixed in that in the PR for `isaiah/desktop-session-recording` and will integrate this with that change.